x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1175,7 +1167,13 @@ private Expression createIfDoesNotAlreadyExist(
             MultiTypeEsField resolvedField,
             List<FieldAttribute> unionFieldAttributes
         ) {
-            var unionFieldAttribute = new FieldAttribute(fa.source(), fa.name(), resolvedField);  // Generates new ID for the field
+            // Generate new ID for the field and suffix it with the data type to maintain unique attribute names.
+            String unionTypedFieldName = SubstituteSurrogates.rawTemporaryName(
```

Comment:
Agree and so I just tried that; it would make sense to move to `esql.core.Attribute`, but then we'd likely want to move `temporaryName` as well; the latter, however, relies on `AggregateFunction` which is outside of `esql.core`.

Maybe let's leave this for another time to not cause too much code movement; this PR needs backporting to 8.15, that could become harder if we also refactor outside of the PR's immediate scope.